Add initial lesson content for lessons when saving outline

Imran92 opened the pull request:
Resolves https://github.com/Automattic/sensei-pro/issues/2498

## Proposed Changes
Enables the lesson outline block and structure saving API to take in and save initialContent for lessons.

More details here https://github.com/Automattic/sensei-pro/pull/2525

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Please check https://github.com/Automattic/sensei-pro/pull/2525

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [x] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [x] All strings are translatable (without concatenation, handles plurals)
- [x] Follows our naming conventions (P6rkRX-4oA-p2)
- [ ] Hooks (p6rkRX-1uS-p2) and functions are documented
- [ ] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [ ] New UIs match the designs
- [ ] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [ ] Legacy courses (course without blocks) are tested
- [x] Code is tested on the minimum supported PHP and WordPress versions
- [ ] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues

Imran92 committed:
Make api receive initialContent prop, sanitize

Imran92 committed:
Save initial content in post and meta

It has to be an a GB markup format, otherwise it won't show. So we wrap it up in a paragraph block's markup.

Imran92 committed:
Also return the initialContent in strcuture

Otherwise the structure in the frontend doesn't match

Imran92 committed:
Add initial content prop in lesson outline block

Imran92 committed:
Return initial content as part of the structure

Imran92 committed:
register the initial content meta field

We need this meta in the frontend in Gutenberg editor. If we don't register the meta, it won't show up in the frontend

Imran92 committed:
Add initial_content to context data

Imran92 committed:
Add initial content data in replacer array

Imran92 committed:
Add sensei description class for content replacement

Imran92 committed:
Update quiz template for content replacement

Imran92 committed:
Update discussion template for content replacement

Imran92 committed:
Update video lesson for content replacement

Imran92 committed:
Test lesson content is set properly

Imran92 committed:
Test that replacer is set properly when called

Imran92 committed:
Remove unnecessary import and clear mocks

Imran92 committed:
Test course structure is saved as expected with initialContenta

Imran92 committed:
Add test for lessons in module

Imran92 committed:
Add tests for content extraction

Imran92 committed:
Merge branch 'trunk' into add/initial-content-functionality-for-lesson

m1r0 commented on includes/class-sensei-lesson.php:
```diff
@@ -5359,6 +5360,39 @@ public function has_sensei_blocks( $lesson = null ) {
+	 * @return boolean Whether the user can edit the meta.
+	 */
+	public function post_meta_auth_callback( $allowed, $meta_key, $post_id ) {
+		return current_user_can( 'edit_post', $post_id );
```
Maybe this is more appropriate?

```suggestion
		return current_user_can( 'edit_lessons', $post_id );
```

Imran92 replied:
Sounds good, I've updated it here https://github.com/Automattic/sensei/pull/7465/commits/661e3456c768e57c0e577d602b2ad1d0a740b22a , Thanks!

Also I've made an update in the Zoom/Meeting and Files to Download patterns here https://github.com/Automattic/sensei/pull/7465/commits/c83ae7eae59f966a36774fdd15c1e1dd682ff361 based on the discussion here https://github.com/Automattic/sensei-pro/pull/2525#issuecomment-1931101649

m1r0 reviewed (dismissed):
Looks good! I've left a small comment, but feel free to merge once that is resolved.

Imran92 committed:
Add a paragraph block in pattens

This helps by giving us a place to insert the lesson content. Also gives the user a place to add their own content

Imran92 committed:
Change permission check

m1r0 reviewed (approved):
🚢
